bulls_and_cows/lib/bulls_and_cows.ex

```diff
@@ -0,0 +1,162 @@
+defmodule BullsAndCows do
+  use Accessible
+
+  @moduledoc """
+  A game of bulls and cows, rules:
+  You need to guess 4 different numbers in definite order.
+  If number is correct but in wrong position, its a cow, if value and position is correct, its a bull.
+
+  Game should be played in iex. First you init game by new_game/0,
+  then you should pass state to make_turn/2 until you win
+  """
+
+  @seed_range 500
+
+  defstruct number_encrypted: "",
+            seed: 0,
+            moves: [],
+            win: false
+
+  @doc """
+  Returns game state
+  """
+  def new_game() do
+    number = generate_number_list()
+    seed = Enum.random(0..@seed_range)
+
+    %__MODULE__{
+      number_encrypted: encrypt(number, seed),
+      seed: seed,
+      moves: [],
+      win: false
+    }
+  end
+
+  @doc """
+  Updates state with your answer, checks win conditions
+
+  ## Examples
+
+      iex> BullsAndCows.make_turn(%BullsAndCows{number_encrypted: "ೞ", seed: 314, moves: [], win: false}, 1234)
+      %BullsAndCows{
+        number_encrypted: "ೞ",
+        seed: 314,
+        moves: [%{answer: 1234, bulls: 0, cows: 1}],
+        win: false
+      }
+
+  """
+  def make_turn(%__MODULE__{win: false} = state, answer) do
+    riddle_list = decrypt(state[:number_encrypted], state[:seed])
+
+    answer_list =
+      answer
+      |> answer_to_list
```

Comment:
```suggestion
    answer_list = answer_to_list(answer)
```
1. Pipes are used when it’s indeed _a pipe_, meaning several transformations. One call is not a pipe, it’s a call.
2. Please use parentheses when calling functions.